server/src/domain/access/access.core.ts

```diff
@@ -0,0 +1,134 @@
+import { BadRequestException } from '@nestjs/common';
+import { AuthUserDto } from '../auth';
+import { IAccessRepository } from './access.repository';
+
+export enum Permission {
+  // ASSET_CREATE = 'asset.create',
```

Comment:
Are we planning to use these commented-out permission later?

Reply:
Yes, I added in everything when i was designing it, but decided not to add permission checks where none previously existed in this PR. In a subsequent one these will be enabled and used.